code/modules/games/cards/lotto.dm

```diff
@@ -0,0 +1,131 @@
+/obj/item/toy/lotto_ticket
+	name = "scratch-off lotto ticket"
+	desc = "A scratch-off lotto ticket."
+	icon = 'icons/obj/toy.dmi'
+	w_class = W_CLASS_TINY
+	var/obj/item/toy/lotto_ticket/revealed = 0
+	var/obj/item/toy/lotto_ticket/iswinner = 0
+	var/obj/item/toy/lotto_ticket/ticket_price
+	var/obj/item/toy/lotto_ticket/winnings = 0
+
+/obj/item/toy/lotto_ticket/New()
+	..()
+	pixel_y = rand(-8, 8) * PIXEL_MULTIPLIER
+	pixel_x = rand(-9, 9) * PIXEL_MULTIPLIER
+
+/obj/item/toy/lotto_ticket/proc/scratch(var/input_price)
+	var/list/prizelist = list(100000,50000,10000,5000,1000,500,250,100,50,20,10,5,4,3,2,1)
+	var/list/problist = list(0.0001, 0.0002, 0.001, 0.002, 0.01, 0.02, 0.04, 0.2, 1, 2.5, 5, 10, 12.5, 17, 20, 25)
+	var/tuning_value = 1/5 //Used to adjust expected values.
+	var/profit = 0
+	if(istype(src,/obj/item/toy/lotto_ticket/supermatter_surprise))
+		input_price = 5	//Sets the prize table equal to the tier-1 card but with a guaranteed win
+	while(profit == 0)
```

Comment:
you can also do `while(!profit)`, though this is purely a matter of style

Also, not sure if this is intended, but this also means that the odds of winning high amounts are MUCH lower than you had on your original proposed table. For example, the odds of winning 2 on a goldrush ticket are not 1 in 5. They are 1 in 5 of the 3 in 4 chance that you didn't win 1. So this is actually 0.2*0.75 -- or 15% chance rather than 20% chance.

Similarly, to win a prize of 3 you need to not win 1 or 2. So that means...
0.75 * 0.8 * 0.166 = ~10% chance, not ~17% chance

This continues all the way up, making larger prizes smaller and smaller.

Reply:
Understood, but I don't see a neutral way of doing this other than randomly assigning each ticket a prize value at creation time (which could be done). Alternate solution could be to continue through the entire list and pick the biggest prize as the final result.

Reply:
After reviewing this further, I don't follow. I iterate from the large prizes to the small intentionally such that if you do win a prize it will always be a big one before a small one. The issue you stated isn't actually an issue. The 17% chance reduced to 10% is only because said reduction is the chance of winning a larger prize. I now disagree that there is a problem here and stand by the implementation

Reply:
Nevermind, I thought you were iterating the other direction